src/modules/navigator/mission_block.cpp

```diff
@@ -224,7 +224,7 @@ MissionBlock::is_mission_item_reached()
 			 * Therefore the item is marked as reached once the system reaches the loiter
 			 * radius (+ some margin). Time inside and turn count is handled elsewhere.
 			 */
-			if (dist >= 0.0f && dist <= _navigator->get_acceptance_radius(fabsf(_mission_item.loiter_radius) * 1.2f)
```

Comment:
This loiter radius can be set by the mission item param3 (https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_LOITER_UNLIM, https://mavlink.io/en/messages/common.html#MAV_CMD_NAV_LOITER_TIME). I think if it is indeed set it should still be taken into account and not the default as you suggest.

Reply:
You are completely right. Please check the newest commit where the default radius is only selected if the mission items loiter radius is invalid.